File: scripts/lib/env-loader.js
import fs from "node:fs";
import path from "node:path";

function normalizeValue(raw) {
  const value = String(raw ?? "").trim();
  if (!value) return "";

  if (
    (value.startsWith('"') && value.endsWith('"')) ||
    (value.startsWith("'") && value.endsWith("'"))
  ) {
    return value.slice(1, -1);
  }

  const commentIndex = value.indexOf(" #");
  if (commentIndex >= 0) {
    return value.slice(0, commentIndex).trim();
  }

  return value;
}

export function parseDotEnv(text) {
  const entries = {};
  for (const rawLine of String(text ?? "").split(/\r?\n/)) {
    const line = rawLine.trim();
    if (!line || line.startsWith("#")) continue;

    const normalized = line.startsWith("export ") ? line.slice(7).trim() : line;
    const separatorIndex = normalized.indexOf("=");
    if (separatorIndex <= 0) continue;

    const key = normalized.slice(0, separatorIndex).trim();
    if (!key) continue;

    const value = normalizeValue(normalized.slice(separatorIndex + 1));
    entries[key] = value;
  }

  return entries;
}

export function loadEnvFileIntoProcess(envPath, { override = false } = {}) {
  if (!envPath || !fs.existsSync(envPath)) {
    return {
      envPath,
      loaded: false,
      count: 0,
      keys: [],
    };
  }

  const payload = parseDotEnv(fs.readFileSync(envPath, "utf8"));
  const keys = [];
  for (const [key, value] of Object.entries(payload)) {
    if (!override && process.env[key] != null) continue;
    process.env[key] = value;
    keys.push(key);
  }

  return {
    envPath,
    loaded: true,
    count: keys.length,
    keys,
  };
}

export function loadProjectEnv(rootDir, options = {}) {
  return loadEnvFileIntoProcess(path.join(rootDir, ".env"), options);
}

File: scripts/lib/env_loader.py
from __future__ import annotations

import os
from pathlib import Path


def _normalize_value(raw: str) -> str:
    value = raw.strip()
    if not value:
        return ""

    if (value.startswith('"') and value.endswith('"')) or (
        value.startswith("'") and value.endswith("'")
    ):
        return value[1:-1]

    comment_index = value.find(" #")
    if comment_index >= 0:
        return value[:comment_index].strip()

    return value


def load_simple_dotenv(env_path: Path, override: bool = False) -> dict[str, str]:
    if not env_path.exists():
        return {}

    loaded: dict[str, str] = {}
    for raw_line in env_path.read_text(encoding="utf-8").splitlines():
        line = raw_line.strip()
        if not line or line.startswith("#"):
            continue

        normalized = line[7:].strip() if line.startswith("export ") else line
        if "=" not in normalized:
            continue

        key, raw_value = normalized.split("=", 1)
        key = key.strip()
        if not key:
            continue

        value = _normalize_value(raw_value)
        if override or key not in os.environ:
            os.environ[key] = value
        loaded[key] = value

    return loaded
